### ISSUE TITLE
QField is broken after update to 3.0.7

### ISSUE THREAD

**Issue by RatInTheField** [opengisch/QField#4792]
### Describe the issue
[A clear and concise description of what the problem is.]
I am sorry, it is not possible to give a clear concise description.

For a very long time, I have been using QField to collect field data. After updating field collection devices with new waypoints, as I have done many times before, I sent the tablets back into the field. The users have not reported any issues, but when I reviewed the data they had collected, fields that are normally populated automatically using aggregate formulae did not contain expected values.

Thinking this was just a glitch, i manually corrected the issues, packaged the data using QFieldSynch and returned it to the field devices. When I checked the functionality of the devices it became clear that it was no longer possible to edit data on them. Message logs report 'Cannot Start editor'

Field devices recently updated to 3.0.7 - Amazonia
qfield synch version 4.6.1
QGIS version  3.34.0-Prizen

Impacts all devices

I am sorry that this report is not very scientific, but I am at a loss to know how to troubleshoot the issue to be able to obtain good data to support your enquiries.

Guidance, suggestions and requests for more information will all be cheerfully accepted.

Thanks

### Reproduction steps
1. [Go to '...']
2. [Click on '...']
3. [Scroll down to '...']

### Expected behavior
[A clear and concise description of what you expected to happen.]

### Observed behavior
[A clear and concise description of what actually happened.]

### Screenshots and GIFs
[If applicable, add screenshots and GIFs to help explain your problem.]

[Please also attach additional files if a specific project/dataset is useful to investigate the problem.]

### Desktop (please complete the following information)
 - OS: [e.g. Windows]
 - QGIS Version [e.g. 3.22.0]
 - QFieldSync Version [e.g.4.0.0]

### Mobile (please complete the following information)
 - OS: [e.g. Android 11]
 - QField version: [Enter QField version here including build number found in the about screen after the release name example 2.0.0 (10200993)]

### Additional information
* Problem started happening recently, didn't happen in an older version of QField: [Yes/No]
* Problem can be reliably reproduced, doesn't happen randomly: [Yes/No]
* Problem happens with all files and projects, not only some files or projects: [Yes/No]

[If the problem happened with QFieldCloud, please add your username and project name.]

**Comment by nirvn** [comment]
@RatInTheField , your description seem to be reporting two different issues, one having to do with default value's aggregate function not returning expected values and another one having to do with "cannot start editing". Am I correct or am I misunderstanding you?

The easiest way to debug this issue would be to provide a sample project (maybe your project with a reduced amount of data?) with steps to reproduce the issue(s) you are facing.

**Comment by RatInTheField** [comment]
[Dovestones_FD_2_19.zip](https://github.com/opengisch/QField/files/13432724/Dovestones_FD_2_19.zip)
Open the project, choose any layer, Waypoint would be a good example, set QField to digitise, create a new waypoint.
The Name field should be populated with a string formatted AAA_A##_#### where A is an alpha character, # is a number and the last 4 numbers are sequential. I use an aggregate formula to set the sequential element. Each new record should have a unique name, but the sequential element is no longer working.
The 'cannot start editing' issue seems to occur when I try to save, edit or delete a record.

Thanks for your speedy response! and please let me know if I need to expand on anything. Its been a very long day and the brain is failing.

**Comment by RatInTheField** [comment]
Interesting. I build all my Qfield apps in Qgis and have not seen this
issue before. I ran checks last night and didn't find the issue in QGIS,
but I will check again.

**Comment by RatInTheField** [comment]
I have now completed some more indepth testing. Its surprising how much easier it is when you're not half asleep and some of the brain is working OK.
Thanks for your input. I appreciate having someone out there who I can bounce ideas off.
Testing revealed local data file corruption probably caused by a system failure earlier yesterday. I have rebuilt the problem layer and so far the issues seem to have been resolved.
Thanks again for providing a calm rational listening ear. Your suggestions were helpful in guiding me to a resolution.
